tools/write_pyi.py

```diff
@@ -99,8 +100,27 @@ def generate_pyi_for_proxy(
                 continue
             meth = getattr(cls, name, None)
             if callable(meth):
+                # check if there are overloads and stub them before the default
+                fn = getattr(cls, name)
+                for ovl in typing.get_overloads(fn):
```

Comment:
Argh, i didn't check the docs. Is it ok to add this RTE? I don't want to limit the maintainer's ability to run this utility and still hand edit the .pyi files. If this is ok to do, i'll add it.

Reply:
I think it's fine since these are dev tools only scripts, and I thinks there is already something similar in a similar one in sqlalchemy for the same reason

Reply:
addressed. There was already a version check requiring 3.9, so i bumped it to 3.11